pkg/catalog/catalog.go

```diff
@@ -1717,50 +1718,116 @@ func (c *Catalog) PrepareExpiredCommits(ctx context.Context, repositoryID string
 	return c.Store.SaveGarbageCollectionCommits(ctx, repository, previousRunID)
 }
 
-func createParquetFile(entries []DBEntry) (string, error) {
-	// TODO: Modify using a run ID generator (https://github.com/treeverse/lakeFS/issues/4469)
-	filename := uuid.New().String()
-	fw, err := local.NewLocalFileWriter(filename)
+type GCUncommittedMark struct {
+	BranchID graveler.BranchID `json:"branch"`
+	Key      graveler.Key      `json:"key"`
+}
+
+func (m *GCUncommittedMark) String() (string, error) {
+	var buf bytes.Buffer
+	encoder := base64.NewEncoder(base64.StdEncoding, &buf)
+	err := json.NewEncoder(encoder).Encode(m)
 	if err != nil {
 		return "", err
 	}
-	defer fw.Close()
-	pw, err := writer.NewParquetWriter(fw, new(DBEntry), int64(len(entries)))
+	result := buf.String()
+	return result, nil
+}
+
+type UncommittedParquetObject struct {
+	PhysicalAddress string `parquet:"name=physical_address, type=BYTE_ARRAY, convertedtype=UTF8, encoding=PLAIN_DICTIONARY"`
+	CreationDate    int64  `parquet:"name=creation_date, type=INT64, convertedtype=INT_64"`
+}
+
+func (c *Catalog) writeUncommittedLocal(ctx context.Context, repository *graveler.RepositoryRecord, fd *os.File, mark GCUncommittedMark) (*GCUncommittedMark, error) {
+	const parallelNum = 1                                                                        // Number of goroutines to handle marshaling of data
+	pw, err := writer.NewParquetWriterFromWriter(fd, new(UncommittedParquetObject), parallelNum) // TODO: Play with np count
 	if err != nil {
-		return "", err
+		return nil, err
 	}
 	// Compression type
 	pw.CompressionType = parquet.CompressionCodec_GZIP
-	for _, entry := range entries {
-		if err = pw.Write(entry); err != nil {
-			return "", err
+
+	const GCMaxUncommittedFileSize = 20 * 1024 * 1024 // 20 MB
+	const periodicCheckSize = 100000
```

Comment:
Since we control the address string:
"data" prefix = 4 bytes
partition id = 20 bytes (xid)
object name = 20 bytes (xid)
timestamp (int64) = 8 bytes

Total per entry ~52 bytes 
Deviation should be around 5 MB

We should also take into consideration that Flushing + Stating the file is a relatively heavy process which will considerably lengthen the operation

WDYT?

Reply:
Notice that for staged entries we do not control the address. 
Either way, your calculation is more accurate than mine -> let's add it as a comment for future reference.